src/extension/serializer.ts

```diff
@@ -512,8 +521,44 @@ export class GrpcSerializer extends SerializerBase {
     )
   }
 
-  private async initParserClient(transport?: GrpcTransport) {
-    this.client = initParserClient(transport ?? (await this.server.transport()))
+  private initClient(): void {
+    // Server options:
+    // (1) pre-existing, started internally by this extension (assuming local execution so it has permissions to start)
+    // (2) pre-existing, started externally, presumably at the this.serverUrl address
+    //   In both cases we need a key & cert (assuming tls) which may reside in:
+    //   (a) this project: '/path/to/projectRoot/tls'
+    //   (b) the runme binary server default: '~/.config/runme/tls'
+    //   (c) other, manually configured
+
+    // assuming 2a, which is the default when starting the server as a result of loading this extension
+    const addTlsConfigIfEnabled = () => {
+      try {
+        if (!getTLSEnabled()) {
+          return {}
+        }
```

Comment:
How about adding a method to KernelServer which returns the pem `async getPem()` and then just call that function here so the logic for getting the PEM is written in a single place inside Kernel?

Reply:
there is already a method in kernelServer with sig `private static async getTLS(tlsDir: string) ` , i can make this public. Adding another method would cause duplication.  are we ok w/ making this public?  I had asked somewhere before if you thought there may be security concerns with this.  Thoughts?

Reply:
Is there a reason to not make it public? If it avoids code duplications what's the downside?

Reply:
It just makes me feel a bit dirty to enable the server to serve up it's certs to anyone that asks for them.  I'll make these methods public and make a note of it in the final PR to have it reviewed.

Reply:
If you don't want to expose the methods. Then the other option is to create a second routine on the KernelServer that can return a connect es transport (in addition) to the gRPC transport. I thought that might lead to more duplicate code but maybe that's cleaner like you say than exposing TLSDir, TLS, and serverUrl.